src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -62,38 +67,74 @@ const SearchPage = () => {
             },
           };
         });
-      setResults(finalResults);
+      setResults({ type: 'author', searchResults: finalResults });
     },
   });
 
   // Hook that will re-render the page only if the state of searchText changes
   useEffect(() => {
-    executeSearch();
-  }, [searchText]);
+    async function searchPosts() {
+      const res = await fetch(`${telescopeUrl}/query?search=${searchText}`);
```

Comment:
Error handling.

Reply:
Also, you need to `encodeURIComponent(searchText)`

Reply:
I checked with express when creating the route before, it will automatically do this

Reply:
This has nothing to do with express.  It will automatically decode it, but you need to create a proper URL.  If I enter "open source" and you put that on the search params, it's not a valid URL any more (you need to encode to "open%20source" or "open+source").

Reply:
Fixed